### ISSUE TITLE
Failed to build centos image

### ISSUE THREAD

**Issue by jscook2345** [apptainer/singularity#2026]
## Version of Singularity:

$ singularity --version
singularity version v3.0.0-alpha.1-578-gdc3529f-dirty

### Expected behavior

A centos image is built using yum without a fatal error.

### Actual behavior

$ sudo singularity build Default.sif examples/centos/Singularity
WARNING: Authentication token file not found : Only pulls of public images will succeed
INFO:    Starting build...
FATAL:   While performing build: conveyor failed to get: While getting bootstrap options: Invalid zypper header, no MirrorURL specified

### Steps to reproduce behavior

I did this from a centos7 virtual machine.

Use the examples/centos/Singularity file from git source tree.

**Comment by ArangoGutierrez** [comment]
the Output says 
```
FATAL: While performing build: conveyor failed to get: While getting bootstrap options: Invalid zypper header, no MirrorURL specified
```
could you share the definition file, to test this issue
thanks

**Comment by jscook2345** [comment]
Its the example in the singularity source tree.

[vagrant@localhost singularity]$ cat examples/centos/Singularity
BootStrap: yum
OSVersion: 7
MirrorURL: http://mirror.centos.org/centos-%{OSVERSION}/%{OSVERSION}/os/$basearch/
Include: yum

%runscript
    echo "This is what happens when you run the container..."


%post
    echo "Hello from inside the container"
    yum -y install vim-minimal

**Comment by jscook2345** [comment]
Looks like some copy / paste issues.

$ ack 'Invalid zypper header'
src/pkg/build/sources/conveyorPacker_yum.go
171:		return fmt.Errorf("Invalid zypper header, no MirrorURL specified")
181:			return fmt.Errorf("Invalid zypper header, OSVersion referenced in mirror but no OSVersion specified")

Why are there zypper comments in the yum conveyorPacker?

**Comment by ArangoGutierrez** [comment]
@ikaneshiro ^^

**Comment by ikaneshiro** [comment]
@jscook2345 The zypper comments are a copy / paste error. Ill clean these up when I circle back to this issue

**Comment by R-Cardenas** [comment]
Hey guys,
I am a new to using singularity. I have exactly the same problem as jscook2345 building a yum image. I understand this was solved in #2064, but cannot follow how it was solved... I am using the remote builder sylabs getting error: 
FATAL:   While performing build: conveyor failed to get: neither yum nor dnf in path. 
But i cant figure whats going on. I hope you can help. 
Best Ryan

**Comment by dtrudg** [comment]
Hi @R-Cardenas - for remote builds this is a different issue I'm afraid. Please see: https://github.com/sylabs/singularity/issues/4615
